hphp/runtime/vm/jit/vasm-instr.h

```diff
@@ -939,6 +948,8 @@ struct incqm { Vptr m; VregSF sf; };
 struct incqmlock { Vptr m; VregSF sf; };
 // mul: s0 * s1 => d, sf
 struct imul { Vreg64 s0, s1, d; VregSF sf; };
+struct mul { Vreg64 s0, s1, d; };
+struct smulh { Vreg64 s0, s1, d; };
```

Comment:
What are `mul{}` and `smulh{}`?  You've designated them as cross-architecture, so you will also need to implement them for x64 and ppc64 (or coordinate with the ppc64 authors, @gut and @lbianc among others, on IRC).  But as far as I can tell, they're only used for the ARM target, so I don't see why they can't be ARM-specific opcodes.